isis/tests/UnitTestImagePolygon.cpp

```diff
@@ -0,0 +1,300 @@
+  FileName labelFile("$ISISROOT/../isis/tests/data/footprintinit/cross.pvl");
+
+  Cube crossCube;
+  crossCube.fromIsd(tempDir.path() + "footprintCube.cub", labelFile, isdFile, "rw");
+
+  ImagePolygon poly;
+  try {
+    poly.Create(crossCube, 100, 100);
+  }
+  catch(IException &e) {
+    QString msg = "Cannot create polygon for [" + crossCube.fileName() + "]";
+    throw IException(IException::Programmer, msg, _FILEINFO_);
+  }
+
+  std::vector<double> lons = {0.000000, 12.467474, 21.544208, 27.693862,
+                              32.115185, 35.471466, 38.127194, 40.290139,
+                              40.345196, 37.640935, 34.997561, 32.400527,
+                              29.817102, 27.267075, 24.742958, 22.241039,
+                              19.755910, 17.283211, 14.791622, 13.394818,
+                              9.905657, 5.816117, 0.912785, 0.000000, 0.000000,
+                              0.000000, 286.909525, 333.822499, 357.984300,
+                              360.000000, 360.000000, 360.000000, 354.859847,
+                              347.059511, 336.340528, 317.180139, 308.824543,
+                              307.775655, 305.511096, 304.451965, 302.574302,
+                              300.053776, 298.105128, 295.343612, 293.012482,
+                              289.800680, 286.909525};
+  std::vector<double> lats = {76.431862, 74.583621, 72.204590, 69.854646,
+                              67.580179, 65.391246, 63.285144, 61.263799,
+                              61.191906, 60.607862, 60.004081, 59.382798,
+                              58.753608, 58.125676, 57.465212, 56.781693,
+                              56.080041, 55.364106, 54.625251, 54.208706,
+                              55.744385, 57.248327, 58.695391, 58.897164,
+                              71.056008, 76.431862, 72.459546, 77.858556,
+                              76.730680, 76.431862, 71.056008, 58.897164,
+                              60.033405, 61.142862, 61.738642, 60.284299,
+                              60.256669, 61.708251, 62.900241, 64.355582,
+                              65.678283, 66.807752, 68.054877, 69.175840,
+                              70.327896, 71.379534, 72.459546};
+
+  geos::geom::CoordinateArraySequence outputCoordArray = geos::geom::CoordinateArraySequence();
+
+  for (auto poly : *(poly.Polys())) {
+    geos::geom::CoordinateArraySequence coordArray = geos::geom::CoordinateArraySequence(*(poly->getCoordinates()));
+    for (int i = 0; i < coordArray.getSize(); i++) {
+      outputCoordArray.add(coordArray.getAt(i));
+    }
+  }
+
+  for (int i = 0; i < outputCoordArray.getSize(); i++) {
+    EXPECT_NEAR(lons[i], outputCoordArray.getAt(i).x, 1e-6);
+    EXPECT_NEAR(lats[i], outputCoordArray.getAt(i).y, 1e-6);
+  }
+}
+
+TEST_F(DefaultCube, UnitTestImagePolygonBoundary) {
+  Pvl footprintLabel;
+
+  std::ifstream isdFile("data/footprintinit/boundary.isd");
```

Comment:
Better to make these new data as part of a fixture and use variables including the process to make them a usable cube.